workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/TypeBlock.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.Gdk.CodeGeneration.CodeWriter.Scopes
+{
+    public class TypeBlock : ScopeBlock, IAnnotatable
+    {
+        internal TypeBlock(string declaration, Action<TypeBlock> populate) : base(declaration)
+        {
+            populate(this);
+        }
+
+        public void Line(string snippet)
+        {
+            Add(new Text(snippet));
+        }
+
+        public void Line(Action<StringBuilder> populate)
+        {
+            var sb = new StringBuilder();
+            populate(sb);
+
+            Add(new Text(sb.ToString()));
+        }
+
+        public void Line(ICollection<string> snippets)
+        {
+            if (snippets.Any())
+            {
+                Add(new TextList(snippets));
+            }
+        }
+
+        public void Initializer(string declaration, Func<IEnumerable<string>> populate)
```

Comment:
What does an initializer on a type block correspond to?

Reply:
discussed offline